http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -153,23 +153,27 @@ void close() {
    * @param errorHandler Invoked if there is a failure to connect.
    * @throws IllegalStateException Thrown if connection is already open (eg: connect called twice).
    * @throws IllegalStateException Thrown if connection has been closed (ie: 'close()' was called)
-   * @return A {@link CompletableFuture} behaving like the {@link CompletableFuture} returned by
-   *     {@link WebSocket.Builder#buildAsync(URI, Listener)} in case of an error.
    */
-  CompletableFuture<WebSocket> connect(
+  @SuppressWarnings("FutureReturnValueIgnored")
+  void connect(final WebSocketConnectionListener listener, final Consumer<String> errorHandler) {
+    connectInternal(listener, errorHandler);
+  }
+
+  @VisibleForTesting
+  CompletableFuture<Void> connectInternal(
       final WebSocketConnectionListener listener, final Consumer<String> errorHandler) {
     this.listener = Preconditions.checkNotNull(listener);
     Preconditions.checkState(client == null);
     Preconditions.checkState(!closed);
 
     return connectAsync()
-        .whenComplete(
-            (webSocket, throwable) -> {
-              if (webSocket != null && throwable == null) {
-                pingSender.start();
-              } else {
-                errorHandler.accept("Failed to connect to: " + serverUri);
-              }
+        .thenRun(pingSender::start)
+        .exceptionally(
+            throwable -> {
+              errorHandler.accept("Failed to connect to: " + serverUri);
+              log.log(
```

Comment:
The error handler should be taking care of any logging that is needed, whether that is a message to a user or a message to the logs.

Reply:
I'm a little bit concerned we might lose the stacktrace in case an actual unintentional error occurs, thoughts?

Reply:
I'm touching code close to this, can see about following up.

The 'severe' logging shows a pop-up to headed game clients, the same for the error handler. We'd wind up showing two error messages to a user along with any others. The stack trace is of some value, but only if a user is reporting the problem via error reporter (stack traces are not useful to users).

If we want to control whether we have a stack trace or not, the error handler perhaps should accept a nullable throwable.